database/migrations/2023_07_13_052204_denormalized_eol_and_add_column_for_explicit_date_to_assets.php

```diff
@@ -66,4 +67,18 @@ public function down()
                 $table->dropColumn('eol_explicit');
         });
     }
+
+    /**
+     * This method returns the correct database expression for either
+     * mysql or sqlite depending on the driver being used.
+     */
+    private function eolUpdateExpression(): Expression
+    {
+        if (DB::getDriverName() === 'sqlite') {
+            return DB::raw("DATE(purchase_date, '+' || (SELECT eol FROM models WHERE models.id = assets.model_id) || ' months')");
```

Comment:
Probably highly unlikely, but should we maybe try the model prefix thing here too?

Reply:
We also have some people using Postgres (also, very unofficially supported) - do we know if this "sqlite" method would work there too? If so, I think maybe we could change this condition to be ` != 'mysql'` instead - so we'd catch sqlite and postgres and anyone else who is running something that's more sql-compliant?

Reply:
Good call...using `DB::getTablePrefix()` there will be a good idea. I'll add it.

I don't know which method, if either, would work for Postgres. The migration has been live for a bit now so I think issues with Postgres would have been raised already. I don't have Postgres installed to test it but I'll check their docs to see if it _looks_ like the original method should have worked for it.

Reply:
We don't have a ton of PgSQL users, but they do exist, even knowing we don't officially support it. 

Some part of me wonders if it would be worth it to just use the "plain" version in the mysql version (assuming there isn't a fancier Laravel way to get these date expressions handled) versus having the if/else.

Reply:
> Some part of me wonders if it would be worth it to just use the "plain" version in the mysql version (assuming there isn't a fancier Laravel way to get these date expressions handled) versus having the if/else.

@snipe can you rephrase? I'm not sure what you mean. 

It would be nice to be able to use `Carbon::now()->addMonths()` but that would mean looping through each asset individually instead of relying on the database to do the update in one command. I don't know another way to do it efficiently...